File: frontend/src/services/adminService.js
import api from "./api";

const adminService = {

    getExerciseById: async (id) => {
    const response = await api.get(`/admin/exercises/${id}`);
    return response.data;
},

    // Get all exercises
    getAllExercises: async ({
        page = 0,
        keyword = "",
        category = "",
        primaryMuscle = "",
        difficulty = "",
        equipment = "",
        exerciseType = "",
        active = ""
    }) => {
        const response = await api.get("/admin/exercises", {
            params: {
                page,
                keyword: keyword || undefined,
                category: category || undefined,
                primaryMuscle: primaryMuscle || undefined,
                difficulty: difficulty || undefined,
                equipment: equipment || undefined,
                exerciseType: exerciseType || undefined,
                active: active === "" ? undefined : active
            }
        });

        return response.data;
    },

    // Get primary muscles by category
    getPrimaryMusclesByCategory: async (category) => {
        const response = await api.get(
            `/exercises/category/${category}/primary-muscles`
        );

        return response.data;
    },

    // Add exercise
    addExercise: async (exerciseData) => {
        const response = await api.post("/admin/exercises", exerciseData);
        return response.data;
    },

    // Update exercise
    updateExercise: async (exerciseId, exerciseData) => {
        const response = await api.patch(
            `/admin/exercises/${exerciseId}`,
            exerciseData
        );
        return response.data;
    },

    // Disable exercise
    disableExercise: async (exerciseId) => {
        const response = await api.patch(
            `/admin/exercises/${exerciseId}/disable`
        );
        return response.data;
    },

    // Enable exercise
    enableExercise: async (exerciseId) => {
        const response = await api.patch(
            `/admin/exercises/${exerciseId}/enable`
        );
        return response.data;
    },

    // Delete user
    deleteUser: async (userId) => {
        const response = await api.delete(
            `/admin/users/${userId}`
        );
        return response.data;
    },

    // Usage statistics
    getUsageStats: async () => {
        const response = await api.get("/admin/usage");
        return response.data;
    },

    // users
    getUsers: async (page = 0) => {
        const response = await api.get("/admin/users", {
            params: { page }
        });
        return response.data;
    },

    disableUser: async (userId) => {
    const response = await api.patch(
        `/admin/users/${userId}/disable`
    );
    return response.data;
},

enableUser: async (userId) => {
    const response = await api.patch(
        `/admin/users/${userId}/enable`
    );
    return response.data;
},


getContactMessages: async ({page = 0,
     status = "",sortBy = "createdAt", direction = "desc"} = {}) => {
    const response = await api.get("/admin/contact-messages", {
        params: {page,status: status || undefined,
            sortBy,direction }
    });
    return response.data;
},
getContactMessageById: async (messageId) => {
    const response = await api.get(
        `/admin/contact-messages/${messageId}`
    );
    return response.data;
},

replyToContactMessage: async (messageId, message) => {
    const response = await api.post(
        `/admin/contact-messages/${messageId}/reply`,
        { message }
    );
    return response.data;
},

updateContactMessageStatus: async (messageId, status) => {
    const response = await api.patch(
        `/admin/contact-messages/${messageId}/status`,
        { status }
    );
    return response.data;
},

deleteContactMessage: async (messageId) => {
    const response = await api.delete(
        `/admin/contact-messages/${messageId}`
    );
    return response.data;
}};



export default adminService;

File: src/main/java/com/sangamesh/Fitsphere/service/AdminService.java
package com.sangamesh.Fitsphere.service;

import com.sangamesh.Fitsphere.dto.admin.AdminExerciseRequestDto;
import com.sangamesh.Fitsphere.dto.admin.AdminExerciseUpdateRequestDto;
import com.sangamesh.Fitsphere.dto.admin.AdminUserSummaryDto;
import com.sangamesh.Fitsphere.dto.contact.ContactMessageResponseDto;
import com.sangamesh.Fitsphere.dto.exercise.ExerciseDetailDto;
import com.sangamesh.Fitsphere.dto.admin.AdminUsageStatsResponseDto;
import com.sangamesh.Fitsphere.dto.exercise.ExerciseSummaryDto;
import com.sangamesh.Fitsphere.enums.*;
import org.springframework.data.domain.Page;

public interface AdminService {



    ExerciseDetailDto addExercise(
            AdminExerciseRequestDto request);

    ExerciseDetailDto updateExercise(Long exerciseId, AdminExerciseUpdateRequestDto request);

    void disableExercise(Long exerciseId);

    void enableExercise(Long exerciseId);

    //users//
    void deleteUser(Long userId);
    void disableUser(Long userId);
    Page<AdminUserSummaryDto> getAllUsers(int page);

    void enableUser(Long userId);

    AdminUsageStatsResponseDto getUsageStats();



    ExerciseDetailDto getExerciseById(Long exerciseId);
    Page<ExerciseSummaryDto> getAllExercises(String keyword, Category category, Muscle primaryMuscle,
                                                    Equipment equipment, Difficulty difficulty, ExerciseType exerciseType,
                                                    Boolean active, String sortBy, String direction, int page);

    //contact
    Page<ContactMessageResponseDto> getContactMessages(ContactMessageStatus status, String sortBy, String direction, int page);
    void replyToContactMessage(Long messageId, String message);
    void updateContactMessageStatus(Long messageId, ContactMessageStatus status);
    void deleteContactMessage(Long messageId);
    ContactMessageResponseDto getContactMessageById(Long messageId);
}